api/src/main/java/io/minio/MinioClient.java

```diff
@@ -896,40 +897,31 @@ public void writeTo(BufferedSink sink) throws IOException {
       // Fix issue #415: No need to compute sha256 if endpoint scheme is HTTPS.
       if (url.isHttps()) {
         sha256Hash = "UNSIGNED-PAYLOAD";
-        if (body instanceof BufferedInputStream) {
-          md5Hash = Digest.md5Hash((BufferedInputStream) body, length);
-        } else if (body instanceof RandomAccessFile) {
-          md5Hash = Digest.md5Hash((RandomAccessFile) body, length);
-        } else if (body instanceof byte[]) {
-          byte[] data = (byte[]) body;
-          md5Hash = Digest.md5Hash(data, length);
+        if (body != null) {
+          md5Hash = Digest.md5Hash(body, length);
         }
       } else {
-        // Fix issue #567: Compute SHA256 hash only.
-        if (body == null) {
-          sha256Hash = Digest.sha256Hash(new byte[0]);
-        } else if (body instanceof BufferedInputStream) {
-          sha256Hash = Digest.sha256Hash((BufferedInputStream) body, length);
-        } else if (body instanceof RandomAccessFile) {
-          sha256Hash = Digest.sha256Hash((RandomAccessFile) body, length);
-        } else if (body instanceof byte[]) {
-          sha256Hash = Digest.sha256Hash((byte[]) body, length);
+        Object data = body;
```

Comment:
you might want to consider ternary operator for conciseness
Object data = (body == null) ? new byte[0] : body;
int len = (body == null) ? 0 : length;